light-client/relayer/src/relayer.rs

```diff
@@ -0,0 +1,267 @@
+        let tracer = tracer!();
+
+        let mut span =
+            block_span_builder(&tracer, "poll_block", self.next_block_index).start(&tracer);
+
+        span.set_attribute(TELEMETRY_BLOCK_INDEX_KEY.i64(self.next_block_index as i64));
+
+        let relevant_burns = Self::check_for_relevant_burns(&block_data.contents().outputs);
+        if relevant_burns.is_empty() {
+            return Ok(());
+        }
+
+        // Try to get signatures from the watcher
+        // Loop until we get enough signatures
+        tracer.in_span("loop_for_signatures", |_cx| {
+            loop {
+                let signatures = self.get_block_signatures(self.next_block_index)?;
+                if signatures.len() >= self.config.min_signatures {
+                    let burned = BurnTx {
+                        block: block_data.block().clone(),
+                        block_contents: block_data.contents().clone(),
+                        signatures,
+                        tx_outs: relevant_burns,
+                    };
+                    self.shared_state().burn_txs.push(burned);
+                    return Ok(());
+                } else {
+                    // We didn't get enough signatures, but let's assume that more are coming soon
+                    // TODO: An alternative here is, actually use a light-client verifier object,
+                    // which is a more correct approach.
+                    log::debug!(
+                        self.logger,
+                        "Did not find a quorum yet for block {}: {} signatures < {} required",
+                        self.next_block_index,
+                        signatures.len(),
+                        self.config.min_signatures
+                    );
+                    std::thread::sleep(Self::POLLING_FREQUENCY);
+                }
+            }
+        })
+    }
+
+    /// Function to match TXOs from a block into interesting vector of
+    /// unspent UTXOs.
+    fn check_for_relevant_burns(outputs: &[TxOut]) -> Vec<TxOut> {
+        // Iterate over each output and filter the results using a parallel iterator.
+        let results: Vec<TxOut> = outputs
+            .into_par_iter()
+            .filter_map(|tx_out| {
+                // View key match against the burn address. If it returns ok, then it's a burn.
+                if tx_out.view_key_match(&burn_address_view_private()).is_ok() {
+                    Some(tx_out.clone())
+                } else {
+                    None
+                }
+            })
+            .collect();
+
+        results
+    }
+
+    fn get_block_signatures(
+        &self,
+        block_index: BlockIndex,
+    ) -> Result<Vec<BlockMetadata>, WatcherDBError> {
+        let block_data_map = self.watcher_db.get_block_data_map(block_index)?;
+        Ok(block_data_map
+            .values()
+            .filter_map(|block_data| block_data.metadata().cloned())
+            .collect())
+    }
```

Comment:
:question: I didn't notice this before. The `get_block_data_map()` appears to have all the information that `ledger.get_block_data()` has. 

Why are we talking to ledger? It seems like an extra external dependency that we might be able to avoid.